src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -247,4 +258,33 @@ public boolean verify(String hostname, SSLSession session) {
     };
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
+
+  private void flushMessageBuffer() {
+    int batchSize = config.getBatchSize();
+    String jsonBatchKey = config.getJsonBatchKey();
+    boolean useJsonBatchKey = !Strings.isNullOrEmpty(jsonBatchKey) && batchSize > 1;
+    boolean shouldWriteJsonAsArray = config.shouldWriteJsonAsArray();
+    switch (config.getMessageFormat()) {
+      case "JSON":
+        String messageBody;
+        if (!shouldWriteJsonAsArray) {
+          messageBody = String.join(config.getDelimiterForMessages(), messageBuffer);
+        } else if (useJsonBatchKey) {
+          messageBody = String.format("{\"%s\":[%s]}", jsonBatchKey, String.join(",", messageBuffer));
+        } else {
+          messageBody = batchSize == 1 ? messageBuffer.get(0) :
```

Comment:
as mentioned in the doc, I think it's better not to have the format dependent on the batch size.

Reply:
Updated to let user use size 1 array